erc/ERCS/erc-4337.md

```diff
@@ -458,7 +468,7 @@ Paymasters facilitate transaction sponsorship, allowing third-party-designed mec
 * No possibility for "passive" paymasters (eg. that accept fees in some ERC-20 token at an exchange rate pulled from an on-chain DEX)
 * Paymasters run the risk of getting griefed, as users could send ops that appear to pay the paymaster but then change their behavior after a block
 
-The paymaster scheme allows a contract to passively pay on users' behalf under arbitrary conditions. It even allows ERC-20 token paymasters to secure a guarantee that they would only need to pay if the user pays them: the paymaster contract can check that there is sufficient approved ERC-20 balance in the `validatePaymasterUserOp` method, and then extract it with `transferFrom` in the `postOp` call; if the op itself transfers out or de-approves too much of the ERC-20s, the inner `postOp` will fail and revert the execution and the outer `postOp` can extract payment (note that because of storage access restrictions the ERC-20 would need to be a wrapper defined within the paymaster itself).
+The paymaster scheme allows a contract to passively pay on users' behalf under arbitrary conditions. for example, a TokenPaymaster uses the precharge refund model. It can precharge the maximum cost with transferFrom in the validatePaymasterUserOp method and then refund the unused portion in the postOp method.
```

Comment:
Why is this change necessary? Also, `TokenPaymaster` is just a specific file name in our repo, I don't think the new text is necessarily better than the old one.

Reply:
I think succinctly writing maybe better:
"Introduction of paymasters allows abstraction of gas, i.e. having a contract, that is not the sender of the transaction, pay for the transaction fees."

Reply:
simplified. it used to try to explain both motivation, implementation and alternate implementations.